L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCTriggerSums.cc

```diff
@@ -0,0 +1,234 @@
+      keV2fC_(conf.getParameter<double>("keV2fC")),
+      fcPerMip_(conf.getParameter<std::vector<double>>("fcPerMip")),
+      layerWeights_(conf.getParameter<std::vector<double>>("layerWeights")),
+      thicknessCorrections_(conf.getParameter<std::vector<double>>("thicknessCorrections")) {}
+
+void HGCalTriggerNtupleHGCTriggerSums::initialize(TTree& tree,
+                                                  const edm::ParameterSet& conf,
+                                                  edm::ConsumesCollector&& collector) {
+  trigger_sums_token_ =
+      collector.consumes<l1t::HGCalTriggerSumsBxCollection>(conf.getParameter<edm::InputTag>("TriggerSums"));
+
+  if (fill_truthmap_)
+    caloparticles_map_token_ =
+        collector.consumes<CaloToCellsMap>(conf.getParameter<edm::InputTag>("caloParticlesToCells"));
+
+  std::string prefix(conf.getUntrackedParameter<std::string>("Prefix", "ts"));
+
+  std::string bname;
+  auto withPrefix([&prefix, &bname](char const* vname) -> char const* {
+    bname = prefix + "_" + vname;
+    return bname.c_str();
+  });
+
+  tree.Branch(withPrefix("n"), &ts_n_, (prefix + "_n/I").c_str());
+  tree.Branch(withPrefix("id"), &ts_id_);
+  tree.Branch(withPrefix("subdet"), &ts_subdet_);
+  tree.Branch(withPrefix("zside"), &ts_side_);
+  tree.Branch(withPrefix("layer"), &ts_layer_);
+  tree.Branch(withPrefix("wafer"), &ts_wafer_);
+  tree.Branch(withPrefix("wafertype"), &ts_wafertype_);
+  tree.Branch(withPrefix("panel_number"), &ts_panel_number_);
+  tree.Branch(withPrefix("panel_sector"), &ts_panel_sector_);
+  tree.Branch(withPrefix("data"), &ts_data_);
+  tree.Branch(withPrefix("pt"), &ts_pt_);
+  tree.Branch(withPrefix("mipPt"), &ts_mipPt_);
+  tree.Branch(withPrefix("energy"), &ts_energy_);
+  tree.Branch(withPrefix("eta"), &ts_eta_);
+  tree.Branch(withPrefix("phi"), &ts_phi_);
+  tree.Branch(withPrefix("x"), &ts_x_);
+  tree.Branch(withPrefix("y"), &ts_y_);
+  tree.Branch(withPrefix("z"), &ts_z_);
+  if (fill_truthmap_)
+    tree.Branch(withPrefix("genparticle_index"), &ts_genparticle_index_);
+}
+
+void HGCalTriggerNtupleHGCTriggerSums::fill(const edm::Event& e, const edm::EventSetup& es) {
+  // retrieve trigger cells
+  edm::Handle<l1t::HGCalTriggerSumsBxCollection> trigger_sums_h;
+  e.getByToken(trigger_sums_token_, trigger_sums_h);
+  const l1t::HGCalTriggerSumsBxCollection& trigger_sums = *trigger_sums_h;
+
+  // retrieve geometry
+  es.get<CaloGeometryRecord>().get(geometry_);
+
+  edm::Handle<CaloToCellsMap> caloparticles_map_h;
+  std::unordered_map<uint32_t, unsigned> module_to_genparticle;
+  if (fill_truthmap_) {
+    e.getByToken(caloparticles_map_token_, caloparticles_map_h);
+    for (auto& keyval : *caloparticles_map_h) {
+      for (auto& tsref : keyval.val)
+        module_to_genparticle.emplace(geometry_->getModuleFromTriggerCell(tsref->detId()),
+                                      keyval.key->g4Tracks().at(0).genpartIndex() - 1);
+    }
+  }
```

Comment:
This can be removed given that there is no static association between trigger cells and trigger sums